lib/babl.rb

```diff
@@ -37,38 +37,41 @@ def self.version
     `#{bin_path} -version`.strip
   end
 
-  def self.client endpoint
-    @clients ||= {}
-    @clients[endpoint || ENV['BABL_ENDPOINT'] || 'default'] ||= begin
-      path = bin_path
-      path = "#{path} -endpoint #{endpoint}" if endpoint
-      Quartz::Client.new(bin_path: path)
-    end
+  def self.client
+    @client ||= Quartz::Client.new(bin_path: bin_path)
   end
 
   def self.module! name, opts = {}
+    res = call! name, opts
+    stdout = Base64.decode64(res["Stdout"])
+    exitcode = res['Exitcode']
+    if exitcode != 0
+      stderr = Base64.decode64(res["Stderr"])
+      raise ModuleError.new(stdout: stdout, stderr: stderr, exitcode: exitcode)
+    end
+    stdout
+  end
+
+  def self.call! name, opts = {}
     params = {'Name' => name}
     if opts[:in]
       params['Stdin'] = Base64.encode64(opts[:in])
     end
     if opts[:env]
       params['Env'] = opts[:env].inject({}) { |h, (k,v)| h[k.to_s] = v.to_s; h }
     end
+    if opts[:endpoint]
```

Comment:
No need to pass `-endpoint` to the binary directly because of the new `BablEndpoint` param